### ISSUE TITLE
Animated Presence like solution for react navigation

### ISSUE THREAD

**Issue by Stevemoretz** [nandorojo/moti#60]
Hi I was thinking, if this could already work or if not would be great to have this feature, so when we close a page or open it in a stack navigator for example the contents would animate.

Well the enter can already happen but the exit is the tricky part, the navigation should be prevented animation should run and then the prevented action should happen only after it.

I've already done this with the old react navigation and old reanimated 1.

But those are now depreciated.
What do you think please let me know.

**Comment by nandorojo** [comment]
Certainly interesting as an idea. I guess you'd have to hook into the navigation state somehow and have it call when the exit completes. 

`AnimatePresence` has a callback for when exits complete: maybe you could call `goBack` there and conditionally render the screen? 🤷🏼‍♂️

**Comment by Stevemoretz** [comment]
Thank for the reply.
I don't know where to use Animated Presence to make this happen, I was thinking if you are interested maybe we could add this to the library itself, it's a feature not found anywhere I've searched for it in a library or something before didn't find anything useful at all.

**Comment by nandorojo** [comment]
Yeah I'm not entirely sure how it would work either. It's an interesting idea but not something I can devote time to. I'm happy to talk through ideas or review a OR if you have them though.

**Comment by Stevemoretz** [comment]
Oh, sure I'm a little busy with other things right now too, maybe I'll do some PRs in the future, when I do this.
